### DISCUSSION TITLE
pdtm does not corectly recognize installed tools

### DISCUSSION THREAD

**Discussion by nth347** [projectdiscovery/pdtm#133] [Q&A] [answered]
### Current Behavior:

I installed some ProjectDiscovery tools by placing their binary files in `/usr/local/bin/`. However, when I run the `pdtm` command, it does not recognize that these tools are installed.

### Expected Behavior:

In my opinion, a better way to check if a tool is installed is to run the tool itself, for example, running `subfinder -version` to verify if Subfinder is installed.

### Steps To Reproduce:

Example: steps to reproduce the behavior:
1. Download pdtm v0.0.6 and run it on Ubuntu 22.04.
2. An incorrect result as follows:
![image](https://user-images.githubusercontent.com/20441996/236211386-e7bca78b-cae9-4a5e-94e4-c8b0c4ea2a5c.png)

**Comment by ehsandeep** [comment]
@nth347 users can have tool installed in multiple ways, for example, go install, brew, apt and more and can have multiple versions of it, which caused issues previously in the past confirming which version/program should be used, instead, we updated pdtm to work with its specific location where it installs the tools (install by pdtm).

The behavior you reported is expected as pdtm does not read/install tools outside of it's default/custom location to avoid updating users' binary arbitrarily.

**Reply by nth347** [reply]
Hi @ehsandeep, does `pdtm` have an option to specify a custom path (e.g., `/usr/local/bin`) to check for installed tools?

**Reply by ehsandeep** [reply, marked as answer]
Not to check, but you can provide a custom location to install the project from pdtm.

**Reply by nth347** [reply]
It's clear now, have a good day @ehsandeep !
